crates/sui-transaction-checks/src/lib.rs

```diff
@@ -88,15 +89,17 @@ mod checked {
         // Runs verifier, which could be expensive.
         check_non_system_packages_to_be_published(transaction, protocol_config, metrics)?;
 
-        let objects = check_input_objects(store, &input_objects, protocol_config)?;
+        let inputs = check_input_objects(store, &input_objects, protocol_config)?;
```

Comment:
it looks to me like this will fail signing if a shared object is deleted. But for client liveness, we need to sign even if the object was deleted (so that it can fail at execution).

Reply:
This should be resolved in the top of commit of this PR!